external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -1920,39 +1920,61 @@ fn parse_control_exp(context: &mut Context) -> Result<(Exp, bool), Box<Diagnosti
             context.tokens.advance()?;
             consume_token(context.tokens, Tok::LParen)?;
             let eb = Box::new(parse_exp(context)?);
-            consume_token(context.tokens, Tok::RParen)?;
-            let (et, ends_in_block) = parse_exp_or_sequence(context)?;
-            let (ef, ends_in_block) = if match_token(context.tokens, Tok::Else)? {
-                let (ef, ends_in_block) = parse_exp_or_sequence(context)?;
-                (Some(Box::new(ef)), ends_in_block)
+            if let Err(diag) = consume_token(context.tokens, Tok::RParen) {
+                context.add_diag(*diag);
+                (
```

Comment:
You could reduce code motion with an early return.

Reply:
Done